src/celengine/completion.h

```diff
@@ -0,0 +1,36 @@
+// completion.h
+//
+// Copyright (C) 2024-present, the Celestia Development Team
+//
+// This program is free software; you can redistribute it and/or
+// modify it under the terms of the GNU General Public License
+// as published by the Free Software Foundation; either version 2
+// of the License, or (at your option) any later version.
+
+#pragma once
+
+#include <functional>
+#include <string>
+#include <celengine/selection.h>
+
+namespace celestia::engine
+{
+
+struct Completion
+{
+    Completion(const std::string& name, const Selection& selection, std::function<Selection()> resolver = nullptr) :
```

Comment:
If I'm reading this correctly, this is an either/or situation - probably should be two separate constructors, one taking a Selection, one taking a resolver.